cmd/ci-operator-prowgen/main.go

```diff
@@ -252,15 +245,8 @@ func main() {
 	config := map[string]*config.Prowgen{}
 	for _, subDir := range args {
 		logger = logger.WithFields(logrus.Fields{"subdir": subDir})
-		if err := generateJobsToDir(opt.fromDir, opt.toDir, subDir, config, opt.resolver); err != nil {
+		if err := opt.generateJobsToDir(subDir, config); err != nil {
 			logger.WithError(err).Fatal("Failed to generate jobs")
 		}
-
-		if opt.ProcessAll() {
-			if err := pruneStaleJobs(opt.toDir, subDir); err != nil {
```

Comment:
Where did this go?

Reply:
Pruning is now done during the write phase (see the new `WriteToDir` in `pkg/jobconfig`, which I still want to clean up a bit) to avoid having to marshal + write + read + unmarshal + marshal + write files.